config/config.go

```diff
@@ -207,23 +207,62 @@ func (c *Config) Save() error {
 
 // resolveAuthConfigForHostname performs the actual auth config resolution
 func (c *Config) resolveAuthConfigForHostname(hostname string) (AuthConfig, error) {
-	// Check credential helpers first
-	if helper, exists := c.CredentialHelpers[hostname]; exists {
-		return c.resolveFromCredentialHelper(helper, hostname)
+	// Generate possible hostname variants for lookup
+	hostVariants := []string{hostname}
+
+	// Strip http:// and https:// prefixes for hostname normalization
+	normalizedHostname := hostname
+	switch {
+	case strings.HasPrefix(hostname, "https://"):
+		normalizedHostname = strings.TrimPrefix(hostname, "https://")
+	case strings.HasPrefix(hostname, "http://"):
+		normalizedHostname = strings.TrimPrefix(hostname, "http://")
 	}
 
-	// Check global credential store
-	if c.CredentialsStore != "" {
-		if authConfig, err := c.resolveFromCredentialHelper(c.CredentialsStore, hostname); err == nil {
+	// Always append both stripped hostname and both scheme variants for cross-scheme lookups
+	if normalizedHostname != hostname {
+		hostVariants = append(hostVariants, normalizedHostname)
+	}
+	hostVariants = append(hostVariants, "https://"+normalizedHostname, "http://"+normalizedHostname)
```

Comment:
It seems strange to me, that we first strip the prefixes, and then add both prefixes again. Can't we transform this into a single idempotent step?